src/StoreMetaUnpacker.cc

```diff
@@ -10,130 +10,80 @@
 
 #include "squid.h"
 #include "base/TextException.h"
-#include "debug/Stream.h"
 #include "defines.h"
 #include "StoreMeta.h"
 #include "StoreMetaUnpacker.h"
 
-int const StoreMetaUnpacker::MinimumBufferLength = sizeof(char) + sizeof(int);
+/* Store::SwapMetaIterator */
 
-/// useful for meta stored in pre-initialized (with zeros) db files
-bool
-StoreMetaUnpacker::isBufferZero()
+Store::SwapMetaIterator::SwapMetaIterator(const void * const start, const void * const end):
+    fieldStart_(static_cast<const char*>(start)),
+    bufEnd_(end)
 {
-    // We could memcmp the entire buffer, but it is probably safe enough
-    // to test a few bytes because if we do not detect a corrupted entry
-    // it is not a big deal. Empty entries are not isBufferSane anyway.
-    const int depth = 10;
-    if (buflen < depth)
-        return false; // cannot be sure enough
-
-    for (int i = 0; i < depth; ++i) {
-        if (buf[i])
-            return false;
-    }
-    return true;
+    sync();
 }
 
-void
-StoreMetaUnpacker::checkBuffer()
+Store::SwapMetaIterator &
+Store::SwapMetaIterator::operator++()
 {
-    assert(buf); // paranoid; already checked in the constructor
-    if (buf[0] != static_cast<char>(STORE_META_OK))
-        throw TexcHere("store entry metadata is corrupted");
-    /*
-     * sanity check on 'buflen' value.  It should be at least big
-     * enough to hold one type and one length.
-     */
-    getBufferLength();
-    if (*hdr_len < MinimumBufferLength)
-        throw TexcHere("store entry metadata is too small");
-    if (*hdr_len > buflen)
-        throw TexcHere("store entry metadata is too big");
+    Assure(fieldStart_ != bufEnd_);
+    fieldStart_ += sizeof(char); // swap meta type
```

Comment:
The whole point of defining the RawSwapMetaType abstraction is to avoid coding its base type everywhere. So...

```suggestion
    fieldStart_ += sizeof(RawSwapMetaType); // swap meta type
```